### PR TITLE
273 transactions payment certificates

### LINKED ISSUE #273: Transactions (disbursement records) and payment certificates
The reactive disclosure section of the CoST IDS has the following element:

> Disbursement records or payment certificates

CoST clarified that disbursement records match the OCDS concept of transactions and that payment certificates are:

> A document certifying that the noted work has been completed and payment to the contractor is approved

## Proposal

Add a `transactions` array to `ContractingProcess/summary` and reuse the OCDS `Transaction` object.

Add a 'paymentCertificate' code to the `documentType` codelist, with the following title and description:

> **Payment certificate**
> A document certifying that the noted work has been completed and payment to the contractor is approved

### PR DESCRIPTION
**Pull request by duncandewhurst**
Closes #273

### CODE REVIEW

**Comment by duncandewhurst** [comment]
@jpmckinney do you know why `test_search` could be failing locally when the PR builds successfully?

```
========================================================================== FAILURES ==========================================================================
______________________________________________________ test_search[en-found \\d+ page\\(s\\) matching] _______________________________________________________

browser = <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>, server = 'http://localhost:8331/', lang = 'en'
regex = 'found \\d+ page\\(s\\) matching'

    @pytest.mark.parametrize('lang,regex', test_search_params)
    def test_search(browser, server, lang, regex):
        browser.get('{}{}'.format(server, lang))
        search_box = browser.find_element_by_id('rtd-search-form').find_element_by_tag_name('input')
        search_box.send_keys('tender\n')
        time.sleep(2)
>       assert re.search(regex, browser.find_element_by_tag_name('body').text)
E       assert None
E        +  where None = <function search at 0x7fa25aecfb80>('found \\d+ page\\(s\\) matching', '            \n\n\n           \n            \n            Language\n            English\n            Español\n        ...ch\nCopyright (c) Open Contracting Partnership, and released under the Apache License 2.0, source available on GitHub.')
E        +    where <function search at 0x7fa25aecfb80> = re.search
E        +    and   '            \n\n\n           \n            \n            Language\n            English\n            Español\n        ...ch\nCopyright (c) Open Contracting Partnership, and released under the Apache License 2.0, source available on GitHub.' = <selenium.webdriver.remote.webelement.WebElement (session="25ffe6f3b2dd015edfbe385464507edd", element="1919841a-53ef-4b4f-84be-0150e37fd572")>.text
E        +      where <selenium.webdriver.remote.webelement.WebElement (session="25ffe6f3b2dd015edfbe385464507edd", element="1919841a-53ef-4b4f-84be-0150e37fd572")> = <bound method WebDriver.find_element_by_tag_name of <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>>('body')
E        +        where <bound method WebDriver.find_element_by_tag_name of <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>> = <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>.find_element_by_tag_name

tests/test_common.py:35: AssertionError
-------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------
127.0.0.1 - - [12/Feb/2021 11:20:33] "GET /en/search/?q=tender&check_keywords=yes&area=default HTTP/1.1" 200 -
127.0.0.1 - - [12/Feb/2021 11:20:33] "GET /en/_static/js/search.js HTTP/1.1" 200 -
__________________________________________________ test_search[es-encontr\xf3 \\d+ p\xe1gina\\(s\\) acorde] __________________________________________________

browser = <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>, server = 'http://localhost:8331/', lang = 'es'
regex = 'encontró \\d+ página\\(s\\) acorde'

    @pytest.mark.parametrize('lang,regex', test_search_params)
    def test_search(browser, server, lang, regex):
        browser.get('{}{}'.format(server, lang))
        search_box = browser.find_element_by_id('rtd-search-form').find_element_by_tag_name('input')
        search_box.send_keys('tender\n')
        time.sleep(2)
>       assert re.search(regex, browser.find_element_by_tag_name('body').text)
E       assert None
E        +  where None = <function search at 0x7fa25aecfb80>('encontró \\d+ página\\(s\\) acorde', '            \n\n\n           \n            \n            Language\n            English\n            Español\n        ...da\nCopyright (c) Open Contracting Partnership, and released under the Apache License 2.0, source available on GitHub.')
E        +    where <function search at 0x7fa25aecfb80> = re.search
E        +    and   '            \n\n\n           \n            \n            Language\n            English\n            Español\n        ...da\nCopyright (c) Open Contracting Partnership, and released under the Apache License 2.0, source available on GitHub.' = <selenium.webdriver.remote.webelement.WebElement (session="25ffe6f3b2dd015edfbe385464507edd", element="7c6c8d8c-f724-40a1-ac11-1b5add2d2b48")>.text
E        +      where <selenium.webdriver.remote.webelement.WebElement (session="25ffe6f3b2dd015edfbe385464507edd", element="7c6c8d8c-f724-40a1-ac11-1b5add2d2b48")> = <bound method WebDriver.find_element_by_tag_name of <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>>('body')
E        +        where <bound method WebDriver.find_element_by_tag_name of <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>> = <selenium.webdriver.chrome.webdriver.WebDriver (session="25ffe6f3b2dd015edfbe385464507edd")>.find_element_by_tag_name

tests/test_common.py:35: AssertionError
-------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------
127.0.0.1 - - [12/Feb/2021 11:20:36] "GET /es/search/?q=tender&check_keywords=yes&area=default HTTP/1.1" 200 -
127.0.0.1 - - [12/Feb/2021 11:20:36] "GET /es/_static/js/search.js HTTP/1.1" 200 -

```

**Comment by jpmckinney** [comment]
The search index (running on the docs server) is only built for a branch once it is pushed to GitHub. It's possible to trigger the index build from your local machine, but I'd say it's too much effort, given that the search test will only fail if the deployment is broken (there's not much you can do in the documentation repositories to cause it to fail, except changing the theme).

Once the PR passes, the local tests are expected to pass.

File: docs/reference/changelog.md [hunk 1]
```diff
@@ -21,40 +22,41 @@
...
+* [#269](https://github.com/open-contracting/infrastructure/issues/269) 'financialAuditReport'.
+* [#272](https://github.com/open-contracting/infrastructure/issues/272) 'qualityAssuranceReport'.
+* [#274](https://github.com/open-contracting/infrastructure/issues/274) 'incorporationCertificate'.
+* [#275](https://github.com/open-contracting/infrastructure/issues/275) 'contractAmendment'.
+* [#273](https://github.com/open-contracting/infrastructure/issues/273) 'paymentCertificate'.
 
 Removed:
 
-* [#268](https://github.com/open-contracting/infrastructure/issues/268) 'finalAudit'
+* [#268](https://github.com/open-contracting/infrastructure/issues/268) 'finalAudit'.
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: For these simple changelog entries, I don't think it's necessary to add a period after the code.

File: docs/reference/changelog.md [hunk 2]
```diff
@@ -65,24 +67,24 @@ Removed:
 
 #### OC4IDS project schema
 
-* [#127](https://github.com/open-contracting/infrastructure/issues/127) - remove the requirement that linked OCDS releases must be provided in release packages containing only one release. Remove recommendation that OCDS releases are cached from schema and add guidance on caching releases from unreliable sources to implementation guidance
-* [#132](https://github.com/open-contracting/infrastructure/issues/132) - add a publicAuthority organization reference field  
-* [#139](https://github.com/open-contracting/infrastructure/issues/139) - update properties of fields in common with OCDS to version [1.1.4](https://standard.open-contracting.org/1.1/en/schema/changelog/#id1)
+* [#127](https://github.com/open-contracting/infrastructure/issues/127) - remove the requirement that linked OCDS releases must be provided in release packages containing only one release. Remove recommendation that OCDS releases are cached from schema and add guidance on caching releases from unreliable sources to implementation guidance.
+* [#132](https://github.com/open-contracting/infrastructure/issues/132) - add a publicAuthority organization reference field  .
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: Remove whitespace before period.

File: schema/project-level/project-schema.json [hunk 3]
```diff
@@ -414,6 +414,15 @@
             "$ref": "#/definitions/Modification"
           },
           "uniqueItems": true
+        },
+        "transactions": {
+          "title": "Transactions",
+          "description": "A list of the spending transactions made against this contracting process.",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]: I don't know if we have it in the style guide yet, but it's preferred not to describe the data type of the field.

```suggestion
          "description": "The spending transactions made against this contracting process.",
```

**Comment by duncandewhurst** [comment]
> The search index (running on the docs server) is only built for a branch once it is pushed to GitHub. It's possible to trigger the index build from your local machine, but I'd say it's too much effort, given that the search test will only fail if the deployment is broken (there's not much you can do in the documentation repositories to cause it to fail, except changing the theme).
>
> Once the PR passes, the local tests are expected to pass.

Good to know, thanks! However, I am still seeing the same failures locally even though the PR passes.

**Comment by jpmckinney** [comment]
Hmm, can you run `pip-sync` (you might need to `pip install pip-tools` first)? You might have an older version of the theme installed.

**Comment by duncandewhurst** [comment]
`pip-sync` did indeed install a new version of the theme (https://github.com/open-contracting/standard_theme/commit/f053ca34989c5f59bfe5e74b686b7a020650562e) but the failures still occur.

File: docs/reference/changelog.md (hunk 1 above)
Comment (duncandewhurst) [reply docs/reference/changelog.md]: Fixed in https://github.com/open-contracting/infrastructure/pull/288/commits/54bc6f2e541f81e412c5aeee58df7798d921e346

File: docs/reference/changelog.md (hunk 2 above)
Comment (duncandewhurst) [reply docs/reference/changelog.md]: Fixed in https://github.com/open-contracting/infrastructure/pull/288/commits/54bc6f2e541f81e412c5aeee58df7798d921e346

**Comment by jpmckinney** [comment]
Did you rebuild the documentation before running the tests? `make` might not recognize the theme had changed, so you can `make clean` first.

**Review by jpmckinney — APPROVED** [review]

**Comment by jpmckinney** [comment]
The tests pass for me locally.

**Comment by duncandewhurst** [comment]
`make_clean` did the trick - thanks! I've added a note to the handbook: https://github.com/open-contracting/standard-development-handbook/pull/238

**Pull request merged**